### ISSUE TITLE
Failed to start proper after 4.1.0, works fine in 4.0.1.

### ISSUE THREAD

**Issue by Eric-Guo** [puma/puma#1905]
### Steps to reproduce

Just upgrade to 4.1.0.

puma.rb file:

```ruby
#!/usr/bin/env puma

directory '/var/www/mixtint/current'
rackup "/var/www/mixtint/current/config.ru"
environment 'production'

tag ''

pidfile "/var/www/mixtint/shared/tmp/pids/puma.pid"
state_path "/var/www/mixtint/shared/tmp/pids/puma.state"
stdout_redirect '/var/www/mixtint/shared/log/puma_access.log', '/var/www/mixtint/shared/log/puma_error.log', true


threads 0,16



bind 'unix:///var/www/mixtint/shared/tmp/sockets/puma.sock'

workers 0





prune_bundler


on_restart do
  puts 'Refreshing Gemfile'
  ENV["BUNDLE_GEMFILE"] = ""
end
```

### Expected behavior

Should start properly.

### Actual behavior

Not start properly.

```log
=== puma startup: 2019-08-12 09:55:50 +0800 ===
Errno::EPIPE: Broken pipe @ io_writev - <STDOUT>
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/events.rb:69:in `write'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/events.rb:69:in `puts'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/events.rb:69:in `log'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/runner.rb:33:in `log'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/single.rb:37:in `stop_blocked'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/launcher.rb:305:in `graceful_stop'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/launcher.rb:399:in `block in setup_signals'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/single.rb:117:in `join'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/single.rb:117:in `run'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/launcher.rb:188:in `run'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/lib/puma/cli.rb:80:in `run'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/gems/puma-4.1.0/bin/puma:10:in `<top (required)>'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/bin/puma:23:in `load'
  /var/www/mixtint/shared/bundle/ruby/2.6.0/bin/puma:23:in `<top (required)>'
```

### System configuration

**Ruby version**: 2.6.3
**Rails version**:  5.2.3/6.0.0.rc2
**Puma version**: 4.1.0

**Comment by nateberkopec** [comment]
Probably related to #1837

**Comment by dentarg** [comment]
> Probably related to #1837

Global `STDOUT`/`STDERR` is used in `Puma::Runner`. Not sure what needs changing though. Maybe @montanalow knows?

https://github.com/puma/puma/blob/a3e18cb3a669b41c4c5e496e5c500abfebb41d08/lib/puma/runner.rb#L114-L138

**Comment by montanalow** [comment]
https://github.com/puma/puma/blob/v4.1.0/lib/puma/launcher.rb#L399

I'm not terribly familiar with this section, but this stack trace indicates SIGTERM is being raised which is killing the processes?

**Comment by Eric-Guo** [comment]
@nafaabout Yes, confirm with #1837 , after rollback 2 line changes, now puma can start properly.

**Comment by montanalow** [comment]
@Eric-Guo your puma.rb works for me with minor path changes on 4.1.0 on OS X using config.ru:

```
run Proc.new { |env| ['200', {'Content-Type' => 'text/html'}, [env]] }
```

```
#!/usr/bin/env puma

directory '.'
rackup "./config.ru"
environment 'production'

tag ''

pidfile "./puma.pid"
state_path "./puma.state"
stdout_redirect './puma_access.log', './puma_error.log', true


threads 0,16



bind 'unix://./puma.sock'

workers 0





prune_bundler


on_restart do
  puts 'Refreshing Gemfile'
  ENV["BUNDLE_GEMFILE"] = ""
end
```

```puma -C puma.rb 
Puma starting in single mode...
* Version 4.1.0 (ruby 2.6.3-p62), codename: Fourth and One
* Min threads: 0, max threads: 16
* Environment: production
* Listening on unix://./puma.sock
Use Ctrl-C to stop
^C- Gracefully stopping, waiting for requests to finish
=== puma shutdown: 2019-08-13 10:13:58 -0700 ===
- Goodbye!
```
What OS are you using?

**Comment by Eric-Guo** [comment]
It's a production server running CentOS 7 and deploys using `capistrano3-puma`, puma.rb file is generated by `capistrano3-puma`

**Comment by idoo** [comment]
have exactly the same issue — rolled back to 4.0.1

**Comment by montanalow** [comment]
1 potential fix is the revert #1837 

The other would be to not `dup` STDOUT and avoid setting `sync = true` to preserve the original intent, and allow the Event pipe to be disconnected during demonization. 

We can add explicit `flush` to `Event` calls that would preserve the current sync behavior of events only, but it seems like there are other Puma use cases (#1906) that currently expect `STDOUT.sync = true`

**Comment by nateberkopec** [comment]
Closing to merge with #1906
